contracts/SimpleSwapFactory.sol

```diff
@@ -14,13 +15,37 @@ contract SimpleSwapFactory {
   /* mapping to keep track of which contracts were deployed by this factory */
   mapping (address => bool) public deployedContracts;
 
+  /* address of the ERC20-token, to be used by the to-be-deployed chequebooks.
+  If 0, deploy SimpleSwap, otherwise ERC20SimpleSwap */
+  address public ERC20Address;
+  constructor(address _ERC20Address) public {
+    ERC20Address = _ERC20Address;
+  }
   /**
   @notice deployes a new SimpleSwap contract
   @param issuer the issuer of cheques for the new chequebook
   @param defaultHardDepositTimeoutDuration duration in seconds which by default will be used to reduce hardDeposit allocations
   */
   function deploySimpleSwap(address payable issuer, uint defaultHardDepositTimeoutDuration)
   public payable returns (address) {
+    if(ERC20Address != address(0)) {
```

Comment:
Debatable if-statement. We could also have two types of factories, just as we have two types of SimpleSwaps

Reply:
Probably better if the token was fixed for a factory.

Reply:
Nevermind. This doesn't exist anymore.